src/coreclr/src/vm/typedesc.cpp

```diff
@@ -40,8 +40,7 @@ BOOL ParamTypeDesc::Verify() {
 
     _ASSERTE(m_TemplateMT.IsNull() || GetTemplateMethodTableInternal()->SanityCheck());
     _ASSERTE(!GetTypeParam().IsNull());
-    BAD_FORMAT_NOTHROW_ASSERT(GetTypeParam().IsTypeDesc() || !GetTypeParam().AsMethodTable()->IsArray());
-    BAD_FORMAT_NOTHROW_ASSERT(CorTypeInfo::IsModifier_NoThrow(GetInternalCorElementType()) ||
```

Comment:
Change this to be a hard assert. We should never come here with bogus element type.